zson/analyzer.go

```diff
@@ -0,0 +1,699 @@
+package zson
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/brimsec/zq/ast"
+	"github.com/brimsec/zq/zng"
+	"github.com/brimsec/zq/zng/resolver"
+)
+
+type Value interface {
+	TypeOf() zng.Type
```

Comment:
it seems like these don't need to be exported?

Reply:
I suspect we will end up using zson.Value in the planner/optimizer/jit-compiler rather than ast (which will have unresolved types).  I think I'll leave it as rather than go through the typing exercise now and then again later...